# Проект спринта 7: Рынок видеоигр

- Автор: Екатерина Мирау
- Дата: 22.02.026

## Цели и задачи проекта

<font color='#777778'>Перечислите цель проекта и задачи, которые вы решаете. Для этого можно использовать описание проекта, но полезнее будет сформулировать основную цель проекта самостоятельно.</font>

**Цель проекта**

Провести подготовительный анализ данных о видеоиграх за период с 2000 по 2013 год включительно, проверить корректность данных и подготовить очищенный и структурированный срез для дальнейшего анализа.

**Задачи проекта**

1. Ознакомиться со структурой данных и оценить их качество:

  - проверить корректность типов данных;

  - выявить пропуски и аномалии;

  - выполнить предобработку при необходимости.


2. Отобрать данные по периоду выпуска игр:
оставить только игры, выпущенные с 2000 по 2013 год включительно.


3. Провести категоризацию игр по оценкам пользователей и экспертов:

   ***Высокая оценка:***

   - пользовательская: от 8 до 10 (включая 10);

   - экспертная: от 80 до 100 (включая 100).

   ***Средняя оценка:***

   - пользовательская: от 3 до 8 (8 не включается);

   - экспертная: от 30 до 80 (80 не включается).

   ***Низкая оценка:***

   - пользовательская: от 0 до 3 (3 не включается);

   - экспертная: от 0 до 30 (30 не включается).


4. Определить топ-7 платформ по количеству игр, выпущенных за выбранный период.

### Описание данных

<font color='#777778'>Здесь приведите описание данных.</font>


Данные из файла **/datasets/new_games.csv** содержат информацию о продажах видеоигр различных жанров и платформ, а также пользовательские и экспертные оценки.

**Описание признаков**

   - **Name** — название игры.

   - **Platform** — платформа, на которой выпущена игра.

   - **Year_of_Release** — год выпуска игры.

   - **Genre** — жанр игры.

   - **NA_sales** — объём продаж в Северной Америке (в миллионах копий).

   - **EU_sales** — объём продаж в Европе (в миллионах копий).

   - **JP_sales** — объём продаж в Японии (в миллионах копий).

   - **Other_sales** — объём продаж в других странах (в миллионах копий).

   - **Critic_Score** — оценка критиков (от 0 до 100).

   - **User_Score** — оценка пользователей (от 0 до 10).

   - **Rating** — возрастной рейтинг, присвоенный организацией **ESRB (Entertainment Software      Rating Board)**, которая классифицирует видеоигры по возрастным категориям.
   

**Категории рейтинга ESRB**

Организация ESRB (Entertainment Software Rating Board) присваивает видеоиграм возрастные категории в зависимости от содержания.

**Основные категории**

- **RP (Rating Pending)** — рейтинг ещё не присвоен. Используется в рекламных материалах до официальной классификации.
Действует с 1994 года.

- **EC (Early Childhood)** — игры для дошкольников.
Не содержат материалов, которые родители могли бы считать неприемлемыми.
Использовался с 1994 по 2018 год.

- **E (Everyone)** — игры для всех возрастов.
Допускается минимальное насилие и лёгкая разговорная лексика.
До 1998 года обозначался как K-A (Kids to Adults).
Действует с 1994 года.

- **E10+ (Everyone 10 and older)** — игры для пользователей от 10 лет.
Возможны более выраженное насилие и умеренно грубая речь.
Введён в 2005 году.

- **T (Teen)** — игры для подростков от 13 лет.
Могут содержать умеренное насилие (включая небольшое количество крови), грубую лексику, юмор и умеренный сексуальный контент.
Действует с 1994 года.

- **M (Mature)** — игры для лиц от 17 лет.
Возможны реалистичное насилие, сцены крови, ужасы, сексуальный контент, частичная нагота и ненормативная лексика.
Действует с 1994 года.

- **AO (Adults Only)** — игры только для взрослых от 18 лет.
Содержат материалы более высокого уровня воздействия, чем категория M.
Действует с 1994 года.

### Содержимое проекта

<font color='#777778'>Перечислите основные шаги проекта или напишите оглавление. Используйте описание проекта, чтобы зафиксировать основные шаги.</font>

---

**1. Введение**

   Цель исследования и описание задачи.

**2. Загрузка и обзор данных**
   
   Импорт библиотек, загрузка датасета, первичный анализ структуры и качества данных.

Предобработка данных

Приведение названий столбцов к формату snake_case

Проверка и корректировка типов данных

Обработка пропусков

Поиск и устранение дубликатов

Фильтрация данных
Отбор игр за период 2000–2013 гг.

Категоризация данных

Разделение игр по пользовательским и экспертным оценкам

Выделение топ-7 платформ по количеству игр

Анализ рынка
Исследование платформ, жанров, региональных продаж и жанра RPG.

Итоговые выводы
Основные результаты и тенденции развития индустрии.



# 2. Загрузка данных и знакомство с ними

- Загрузите необходимые библиотеки Python и данные датасета `/datasets/new_games.csv`.


- Познакомьтесь с данными: выведите первые строки и результат метода `info()`.


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('/datasets/new_games.csv')

In [3]:
# Вывод первых строк датасета
df.head()

,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


In [4]:
# Общая информация о данных
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  object 
 1   Platform         16956 non-null  object 
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  object 
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  object 
 6   JP sales         16956 non-null  object 
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  object 
 10  Rating           10085 non-null  object 
dtypes: float64(4), object(7)
memory usage: 1.4+ MB


- Сделайте вывод о полученных данных: данные какого объёма вам предоставили, соответствуют ли они описанию, встречаются ли в них пропуски, используются ли верные типы данных.
- Отметьте другие особенности данных, которые вы обнаружили и на которые стоит обратить внимание при предобработке. Например, вы можете проверить названия столбцов: все ли названия отражают содержимое данных и прописаны в удобном для работы виде.

<font color='#777778'>Используйте ячейки типа Markdown для промежуточных выводов и расширенных комментариев к действиям с данными. </font>

**Первичный анализ данных**

В датасете содержится 16 956 строк и 11 столбцов. Объём данных достаточный для проведения исследования. 

**Соответствие описанию**

Представлены все заявленные переменные: название игры, платформа, год выпуска, жанр, региональные продажи, оценки критиков и пользователей, а также возрастной рейтинг ESRB. Данные соответствуют документации.

**Наличие пропусков**

В данных обнаружены пропущенные значения:

- **Name** — 2 пропуска

- **Year of Release** — 275 пропусков

- **Genre** — 2 пропуска

- **Critic Score** — 8714 пропуска, это более половины значений

- **User Score** — 6804 пропуска, тоже значительное количество 

- **Rating** — 6871 пропуска, это около 40% значений 

Особенно заметны пропуски в оценках и рейтинге, что может быть связано с отсутствием рецензий или неполнотой исторических данных.

**Корректность типов данных**

Обнаружены несоответствия типов:

- **EU sales и JP sales** имеют тип **object**, хотя должны быть числовыми.

- **User Score** имеет тип **object**, вероятно из-за наличия строковых значений (например, текстовых маркеров вместо чисел).

- **Year of Release** имеет тип **float64**, что связано с пропусками; после их обработки столбец можно привести к целочисленному типу.

**Дополнительные особенности**

- Названия столбцов содержат пробелы и разный стиль написания (например, Year of Release,     NA sales). Их следует привести к формату snake_case для удобства работы.

- Необходимо проверить категориальные данные (жанры, платформы, рейтинг) на наличие неявных   дубликатов и различий в написании.

- Требуется анализ природы пропусков перед их обработкой.

**Общий вывод**

Данные подходят для проведения анализа, однако требуют предварительной очистки: приведения типов данных, обработки пропусков, нормализации названий столбцов и проверки дубликатов. После предобработки можно переходить к формированию аналитического среза за период 2000–2013 гг.

---

## Проверка ошибок в данных и их предобработка


###  Названия, или метки, столбцов датафрейма

- Выведите на экран названия всех столбцов датафрейма и проверьте их стиль написания.
- Приведите все столбцы к стилю snake case. Названия должны быть в нижнем регистре, а вместо пробелов — подчёркивания.

In [5]:
# Выводим названиявсех столбцов датафрейма
df.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='object')

In [6]:
# Приводим все столбцы к стилю snake case
df.columns = (
    df.columns
    .str.lower()        # нижний регистр
    .str.replace(' ', '_')  # замена пробелов на _
)

df.columns

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='object')

### Типы данных

- Если встречаются некорректные типы данных, предположите их причины.
- При необходимости проведите преобразование типов данных. Помните, что столбцы с числовыми данными и пропусками нельзя преобразовать к типу `int64`. Сначала вам понадобится обработать пропуски, а затем преобразовать типы данных.

**Предположения о причинах некорректных типов**

*eu_sales*, *jp_sales* имеют тип *object*, хотя это продажи. Скорее всего, в столбцах встречаются строковые значения.

*user_score* тоже *object*, потому что кроме чисел часто встречаются строковые значения, например unknown и/или пропуски.

*year_of_release* — *float*, потому что есть пропуски. 

- В числовых столбцах могут встретиться строковые значения, например `unknown` или другие. Приводите такие столбцы к числовому типу данных, заменив строковые значения на пропуски.

In [7]:
# Смотрим типы до преобразований
df.dtypes

name                object
platform            object
year_of_release    float64
genre               object
na_sales           float64
eu_sales            object
jp_sales            object
other_sales        float64
critic_score       float64
user_score          object
rating              object
dtype: object

In [8]:
# Смотрим уникальные значения в eu_sales 
df['eu_sales'].unique()

array(['28.96', '3.58', '12.76', '10.93', '8.89', '2.26', '9.14', '9.18',
       '6.94', '0.63', '10.95', '7.47', '6.18', '8.03', '4.89', '8.49',
       '9.09', '0.4', '3.75', '9.2', '4.46', '2.71', '3.44', '5.14',
       '5.49', '3.9', '5.35', '3.17', '5.09', '4.24', '5.04', '5.86',
       '3.68', '4.19', '5.73', '3.59', '4.51', '2.55', '4.02', '4.37',
       '6.31', '3.45', '2.81', '2.85', '3.49', '0.01', '3.35', '2.04',
       '3.07', '3.87', '3.0', '4.82', '3.64', '2.15', '3.69', '2.65',
       '2.56', '3.11', '3.14', '1.94', '1.95', '2.47', '2.28', '3.42',
       '3.63', '2.36', '1.71', '1.85', '2.79', '1.24', '6.12', '1.53',
       '3.47', '2.24', '5.01', '2.01', '1.72', '2.07', '6.42', '3.86',
       '0.45', '3.48', '1.89', '5.75', '2.17', '1.37', '2.35', '1.18',
       '2.11', '1.88', '2.83', '2.99', '2.89', '3.27', '2.22', '2.14',
       '1.45', '1.75', '1.04', '1.77', '3.02', '2.75', '2.16', '1.9',
       '2.59', '2.2', '4.3', '0.93', '2.53', '2.52', '1.79', '1.3', '2.6',
   

In [9]:
# Смотрим уникальные значения в jp_sales 
df['jp_sales'].unique()

array(['3.77', '6.81', '3.79', '3.28', '10.22', '4.22', '6.5', '2.93',
       '4.7', '0.28', '1.93', '4.13', '7.2', '3.6', '0.24', '2.53',
       '0.98', '0.41', '3.54', '4.16', '6.04', '4.18', '3.84', '0.06',
       '0.47', '5.38', '5.32', '5.65', '1.87', '0.13', '3.12', '0.36',
       '0.11', '4.35', '0.65', '0.07', '0.08', '0.49', '0.3', '2.66',
       '2.69', '0.48', '0.38', '5.33', '1.91', '3.96', '3.1', '1.1',
       '1.2', '0.14', '2.54', '2.14', '0.81', '2.12', '0.44', '3.15',
       '1.25', '0.04', '0.0', '2.47', '2.23', '1.69', '0.01', '3.0',
       '0.02', '4.39', '1.98', '0.1', '3.81', '0.05', '2.49', '1.58',
       '3.14', '2.73', '0.66', '0.22', '3.63', '1.45', '1.31', '2.43',
       '0.7', '0.35', '1.4', '0.6', '2.26', '1.42', '1.28', '1.39',
       '0.87', '0.17', '0.94', '0.19', '0.21', '1.6', '0.16', '1.03',
       '0.25', '2.06', '1.49', '1.29', '0.09', '2.87', '0.03', '0.78',
       '0.83', '2.33', '2.02', '1.36', '1.81', '1.97', '0.91', '0.99',
       '0.95', '2.0'

In [10]:
# Смотрим уникальные значения в user_score 
df['user_score'].unique()

array(['8', nan, '8.3', '8.5', '6.6', '8.4', '8.6', '7.7', '6.3', '7.4',
       '8.2', '9', '7.9', '8.1', '8.7', '7.1', '3.4', '5.3', '4.8', '3.2',
       '8.9', '6.4', '7.8', '7.5', '2.6', '7.2', '9.2', '7', '7.3', '4.3',
       '7.6', '5.7', '5', '9.1', '6.5', 'tbd', '8.8', '6.9', '9.4', '6.8',
       '6.1', '6.7', '5.4', '4', '4.9', '4.5', '9.3', '6.2', '4.2', '6',
       '3.7', '4.1', '5.8', '5.6', '5.5', '4.4', '4.6', '5.9', '3.9',
       '3.1', '2.9', '5.2', '3.3', '4.7', '5.1', '3.5', '2.5', '1.9', '3',
       '2.7', '2.2', '2', '9.5', '2.1', '3.6', '2.8', '1.8', '3.8', '0',
       '1.6', '9.6', '2.4', '1.7', '1.1', '0.3', '1.5', '0.7', '1.2',
       '2.3', '0.5', '1.3', '0.2', '0.6', '1.4', '0.9', '1', '9.7'],
      dtype=object)

In [11]:
# Приводим столбцы с продажами и пользовательской оценкой к числовому типу
# Все нечисловые значения станут NaN
cols_to_numeric = ['eu_sales', 'jp_sales', 'user_score']

for col in cols_to_numeric:
    df[col] = pd.to_numeric(df[col], errors='coerce')

In [12]:
# проверяем, что типы поменялись
df[cols_to_numeric].dtypes

eu_sales      float64
jp_sales      float64
user_score    float64
dtype: object

In [13]:
# сколько стало NaN после преобразований
df[cols_to_numeric].isna().sum()

eu_sales         6
jp_sales         4
user_score    9268
dtype: int64

### Наличие пропусков в данных

- Посчитайте количество пропусков в каждом столбце в абсолютных и относительных значениях.


In [14]:
# абсолютное количество пропусков
df.isna().sum()

name                  2
platform              0
year_of_release     275
genre                 2
na_sales              0
eu_sales              6
jp_sales              4
other_sales           0
critic_score       8714
user_score         9268
rating             6871
dtype: int64

In [15]:
# относительное количество пропусков 
(df.isna().mean() * 100).round(2)

name                0.01
platform            0.00
year_of_release     1.62
genre               0.01
na_sales            0.00
eu_sales            0.04
jp_sales            0.02
other_sales         0.00
critic_score       51.39
user_score         54.66
rating             40.52
dtype: float64

- Изучите данные с пропущенными значениями. Напишите промежуточный вывод: для каких столбцов характерны пропуски и сколько их. Предположите, почему пропуски могли возникнуть. Укажите, какие действия с этими данными можно сделать и почему.


**Промежуточный вывод по пропускам**

Наибольшее количество пропусков наблюдается в столбцах:

- **user_score** — 9268 пропусков (54,66%)

- **critic_score** — 8714 пропусков (51,39%)

- **rating** — 6871 пропуск (40,52%)

Это может быть связано с тем, что не для всех игр доступны пользовательские и экспертные оценки, а также возрастной рейтинг. Особенно это характерно для старых игр или проектов с небольшим охватом.

В столбце **ear_of_release** пропущено 275 значений (1,62%).
В столбцах **name, genre, eu_sales, jp_sales** пропуски незначительны (менее 0,05%).

**Возможные действия**

- Пропуски в sales можно заменить на средние значения по платформе и году выпуска (если потребуется для анализа).

- Пропуски в оценках и рейтинге целесообразно оставить как NaN, так как их невозможно корректно восстановить без дополнительных источников.

- Строки с пропусками в name и genre можно удалить из-за их незначительного количества.

- Пропуски в year_of_release требуют дополнительного анализа: возможна либо замена, либо удаление строк.

В целом основные проблемы сосредоточены в оценках и рейтинге, что важно учитывать при дальнейшем анализе.

- Обработайте пропущенные значения. Для каждого случая вы можете выбрать оптимальный, на ваш взгляд, вариант: заменить на определённое значение, оставить как есть или удалить.
- Если вы решите заменить пропуски на значение-индикатор, то убедитесь, что предложенное значение не может быть использовано в данных.
- Если вы нашли пропуски в данных с количеством проданных копий игры в том или ином регионе, их можно заменить на среднее значение в зависимости от названия платформы и года выхода игры.

In [16]:
# name, genre - количество незначительное, удаляем строки
df = df.dropna(subset=['name', 'genre'])

In [17]:
# Год важен для анализа, поэтому удаляем строки с пропущенным годом, чтобы не искажать результат
df = df.dropna(subset=['year_of_release'])

In [18]:
# Теперь приводим год к целочисленному типу
df['year_of_release'] = df['year_of_release'].astype('Int64')

In [19]:
# Пропуски в eu_sales, jp_sales, заменяем на средние значения по платформе и году выпуска:  
sales_cols = ['eu_sales', 'jp_sales']
for col in sales_cols:
    df[col] = df[col].fillna(
        df.groupby(['platform', 'year_of_release'])[col].transform('mean'))

In [20]:
# Заменяем пропуски в rating на индикатор "no_rating"
df['rating'] = df['rating'].fillna('no_rating')

In [21]:
df.isna().sum()

name                  0
platform              0
year_of_release       0
genre                 0
na_sales              0
eu_sales              0
jp_sales              0
other_sales           0
critic_score       8594
user_score         9121
rating                0
dtype: int64

Оценки оставлены без заполнения для сохранения корректности анализа.

### Явные и неявные дубликаты в данных

- Изучите уникальные значения в категориальных данных, например с названиями жанра игры, платформы, рейтинга и года выпуска. Проверьте, встречаются ли среди данных неявные дубликаты, связанные с опечатками или разным способом написания.
- При необходимости проведите нормализацию данных с текстовыми значениями. Названия или жанры игр можно привести к нижнему регистру, а названия рейтинга — к верхнему.

In [22]:
# Смотрим количество уникальных значений
df[['platform', 'genre', 'rating']].nunique()

platform    31
genre       24
rating       9
dtype: int64

In [23]:
# Смотрим сами уникальные значения
print("Platforms:\n", df['platform'].unique())
print("\nGenres:\n", df['genre'].unique())
print("\nRatings:\n", df['rating'].unique())

Platforms:
 ['Wii' 'NES' 'GB' 'DS' 'X360' 'PS3' 'PS2' 'SNES' 'GBA' 'PS4' '3DS' 'N64'
 'PS' 'XB' 'PC' '2600' 'PSP' 'XOne' 'WiiU' 'GC' 'GEN' 'DC' 'PSV' 'SAT'
 'SCD' 'WS' 'NG' 'TG16' '3DO' 'GG' 'PCFX']

Genres:
 ['Sports' 'Platform' 'Racing' 'Role-Playing' 'Puzzle' 'Misc' 'Shooter'
 'Simulation' 'Action' 'Fighting' 'Adventure' 'Strategy' 'MISC'
 'ROLE-PLAYING' 'RACING' 'ACTION' 'SHOOTER' 'FIGHTING' 'SPORTS' 'PLATFORM'
 'ADVENTURE' 'SIMULATION' 'PUZZLE' 'STRATEGY']

Ratings:
 ['E' 'no_rating' 'M' 'T' 'E10+' 'K-A' 'AO' 'EC' 'RP']


In [24]:
# Приводим названия игр и жанры к нижнему регистру
df['name'] = df['name'].str.lower()
df['genre'] = df['genre'].str.lower()

In [25]:
# Убираем лишние пробелы
df['name'] = df['name'].str.strip()
df['genre'] = df['genre'].str.strip()
df['platform'] = df['platform'].str.strip()
df['rating'] = df['rating'].str.strip()

In [26]:
# Рейтинг приводим к верхнему регистру
df['rating'] = df['rating'].str.upper()

- После того как нормализуете данные и устраните неявные дубликаты, проверьте наличие явных дубликатов в данных.

In [27]:
# Проверяем наличие явных дубликатов
df.duplicated().sum()

235

In [28]:
# Удаляем явные дубликаты
df = df.drop_duplicates()

In [29]:
# Проверяем повторно
df.duplicated().sum()

0

- Напишите промежуточный вывод: укажите количество найденных дубликатов и действия по их обработке.

**Промежуточный вывод по неявным дубликатам**

В категориальных столбцах были обнаружены неявные дубликаты, связанные с различием регистра 
(например, Sports и SPORTS, Action и ACTION).

Для устранения проблемы текстовые данные были нормализованы:
- названия игр и жанры приведены к нижнему регистру;
- рейтинг приведён к верхнему регистру;
- удалены лишние пробелы.

После нормализации количество уникальных значений сократилось, 
что подтверждает устранение неявных дубликатов.

**Промежуточный вывод по явным дубликатам**

После нормализации текстовых данных была выполнена проверка на наличие явных дубликатов. 
Обнаружено **235** полностью совпадающих строк.

Дубликаты были удалены с помощью метода drop_duplicates(). 
Повторная проверка показала отсутствие явных дубликатов в датасете.

- В процессе подготовки данных вы могли что-либо удалять, например строки с пропусками или ошибками, дубликаты и прочее. В этом случае посчитайте количество удалённых строк в абсолютном и относительном значениях.

In [30]:
rows_before = 16956  # исходное количество строк
rows_after = len(df)

removed_rows = rows_before - rows_after
removed_percent = round((removed_rows / rows_before) * 100, 2)

removed_rows, removed_percent

(512, 3.02)

**Итог по удалённым строкам**

В процессе предобработки данных были удалены строки с пропусками в ключевых признаках 
и выявленные явные дубликаты.

Всего было удалено 512 строк, что составляет 3,02% от исходного объёма данных.

Удаление не оказало существенного влияния на общий объём датасета, 
но позволило повысить качество данных и корректность дальнейшего анализа.

- После проведения предобработки данных напишите общий промежуточный вывод.

**Общий промежуточный вывод по предобработке данных**

- В ходе предобработки данные были приведены к корректному и удобному для анализа виду.

- Названия столбцов приведены к формату snake_case.

- Исправлены типы данных: числовые столбцы приведены к числовому формату, строковые значения заменены на NaN.

- Проанализированы и обработаны пропущенные значения: удалены строки с пропусками в ключевых признаках, для части данных применена замена.

- Устранены неявные дубликаты путём нормализации текстовых данных (приведение регистра, удаление лишних пробелов).

- Обнаружено и удалено 235 явных дубликатов.

- В общей сложности удалено 512 строк (3,02% от исходного объёма данных).

---

## Фильтрация данных

Коллеги хотят изучить историю продаж игр в начале XXI века, и их интересует период с 2000 по 2013 год включительно. Отберите данные по этому показателю. Сохраните новый срез данных в отдельном датафрейме, например `df_actual`.

In [31]:
# Фильтрация данных по периоду 2000–2013 гг.
df_actual = df[
    (df['year_of_release'] >= 2000) &
    (df['year_of_release'] <= 2013)
].copy()

df_actual.shape

(12781, 11)

---

## Категоризация данных
    
Проведите категоризацию данных:
- Разделите все игры по оценкам пользователей и выделите такие категории: высокая оценка (от 8 до 10 включительно), средняя оценка (от 3 до 8, не включая правую границу интервала) и низкая оценка (от 0 до 3, не включая правую границу интервала).

In [32]:
df_actual['user_score_category'] = pd.cut(
    df_actual['user_score'],
    bins=[-1, 0, 3, 8, 10],
    labels=['без рейтинга', 'низкая', 'средняя', 'высокая'],
    right=False,
    include_lowest=True
)

In [33]:
df_actual['user_score_category'].value_counts()

средняя         4081
высокая         2286
низкая           116
без рейтинга       0
Name: user_score_category, dtype: int64

- Разделите все игры по оценкам критиков и выделите такие категории: высокая оценка (от 80 до 100 включительно), средняя оценка (от 30 до 80, не включая правую границу интервала) и низкая оценка (от 0 до 30, не включая правую границу интервала).

In [34]:
df_actual['critic_score_category'] = pd.cut(
    df_actual['critic_score'],
    bins=[-1, 0, 30, 80, 100], 
    labels=['без рейтинга', 'низкая', 'средняя', 'высокая'],
    right=False,
    include_lowest=True)

In [35]:
df_actual['critic_score_category'].value_counts()

средняя         5422
высокая         1692
низкая            55
без рейтинга       0
Name: critic_score_category, dtype: int64

- После категоризации данных проверьте результат: сгруппируйте данные по выделенным категориям и посчитайте количество игр в каждой категории.

In [36]:
# Группировка по пользовательским категориям
df_actual.groupby('user_score_category')['name'].count()

user_score_category
без рейтинга       0
низкая           116
средняя         4081
высокая         2286
Name: name, dtype: int64

In [37]:
# Группировка по категориям критиков
df_actual.groupby('critic_score_category')['name'].count()

critic_score_category
без рейтинга       0
низкая            55
средняя         5422
высокая         1692
Name: name, dtype: int64

- Выделите топ-7 платформ по количеству игр, выпущенных за весь актуальный период.

In [38]:
# Топ-7 платформ по количеству выпущенных игр
df_actual['platform'].value_counts().head(7)

PS2     2127
DS      2120
Wii     1275
PSP     1180
X360    1121
PS3     1087
GBA      811
Name: platform, dtype: int64

---

## Итоговый вывод

В конце напишите основной вывод и отразите, какую работу проделали. Не забудьте указать описание среза данных и новых полей, которые добавили в исходный датасет.

В ходе проекта были проанализированы данные о продажах видеоигр. На первом этапе выполнена предобработка: приведены названия столбцов к формату **snake_case**, исправлены типы данных, строковые значения в числовых столбцах заменены на пропуски, обработаны отсутствующие значения и устранены явные и неявные дубликаты. Это позволило привести данные к корректному и удобному для анализа виду.

Далее был сформирован актуальный срез данных **за период 2000–2013 гг.,** соответствующий цели исследования.

**В рамках анализа:**

Игры были распределены по категориям пользовательских оценок **(низкая, средняя, высокая).**

Игры также были распределены по категориям оценок критиков, включая отдельную категорию для игр **без рейтинга.**

Определены **топ-7 платформ** по количеству выпущенных игр за рассматриваемый период.

В результате проделанной работы сформирован структурированный и очищенный датасет, дополненный новыми категориальными признаками **(user_score_category, critic_score_category)**. Подготовленные данные могут быть использованы для дальнейшего анализа развития рынка видеоигр и оценки динамики платформ и качества игр в выбранный период.